internal/services/elasticsan/elastic_san_volume_snapshot_data_source_test.go

```diff
@@ -0,0 +1,114 @@
+package elasticsan_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type ElasticSANVolumeSnapshotDataSource struct{}
+
+// https://github.com/hashicorp/terraform-provider-azurerm/pull/25372#issuecomment-2022105240
+// Elastic SAN Volume Snapshot is context-based and should not be regarded as the infrastructure managed by Terraform
+// so we only onboard this as a data source instead of a resource. The acctest relys on Azure CLI to create/delete the snapshot.
+func TestAccElasticSANVolumeSnapshotDataSource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_elastic_san_volume_snapshot", "test")
+	d := ElasticSANVolumeSnapshotDataSource{}
+
+	const SnapshotTestRunEnv = "ARM_TEST_ELASTIC_SAN_VOLUME_SNAPSHOT_RUN"
+
+	if os.Getenv(SnapshotTestRunEnv) == "" {
+		t.Skipf("skip the test as one or more of below environment variables are not specified: %q", SnapshotTestRunEnv)
+	}
```

Comment:
We should add the creation of the Snapshot as a test step in here instead of using the `local-exec` provisioner to run `az cli`. That way we don't need to set an environment variable to run this test. Here's an example of where we provision resources "outside" of Terraform in tests:

https://github.com/hashicorp/terraform-provider-azurerm/blob/d0f0e8019e683345671442e2d0b64053796a01b2/internal/services/containers/kubernetes_node_pool_snapshot_data_source_test.go#L23-L58

Reply:
changed